modules/llrt_crypto/src/subtle/crypto_key.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use std::rc::Rc;
+
+use rquickjs::{class::Trace, Array, Class, Ctx, JsLifetime, Result, Value};
+
+#[rquickjs::class]
+#[derive(Clone, Trace, rquickjs::JsLifetime)]
+pub struct CryptoKey<'js> {
```

Comment:
We don't want to store the JS values in here but the rust structs directly. This is because storing the JS values allows access by reference and make them modifiable. Instead we should convert them to JS values on access.
For instance, in the algorithm getter we should convert KeyGenAlgorithm into JS but only on access. This also offers better performance as we can remove converting to and from rust as the key is likely passed to another function. Same methodology goes for all other CryptoAPI structures that should return non-modifiable data:
See https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API

Reply:
Does this mean an implementation like the [CryptoKey internal slots](https://w3c.github.io/webcrypto/#cryptokey-interface-internal-slots) from the Web Cryptography API?

Currently, only the handle is implemented internally in this way, but the idea is to implement it so that other CryptoKey variables are also managed internally.

Reply:
Since the algorithm is an object we pass from user space (JavaScript) we have two options. One is to convert that to an internal rust representation and store it (I think there is a proc macro for this as well) . The other is to deep clone it when access it (not preferable). The reason for this is that we should not be able to change values inside of this object after it has been passed to generate key. Remember that .clone() on a Value or Object in rquickjs returns a reference to the same object.

Reply:
@nabetti1720 please let me know if you wan't support for this specific task. I can check out your branch and push the required changes for this change request 🥇

Reply:
@richarddavison, Thank you for your support!

I thought this part would be a little difficult to fix. If you have the time, could you please fix this branch? I'd love to learn from your fixes.